src/packages/auth/src/authentication/methods/utils.ts

```diff
@@ -70,14 +77,63 @@ export const runAfterHooks = async <D, H = CredentialInsertInput | CredentialUpd
 	return hookEntities;
 };
 
-export const maskSensitiveValuesForLogging = (
+const obfuscateSensitiveValues = (
+	variables: VariableValues | undefined,
+	nodeNamesToMask: Set<string>
+) => {
+	try {
+		if (!variables) return variables;
+
+		const obfuscateNode = (key: string, value: any): any => {
+			if (nodeNamesToMask.has(key)) {
+				return '********';
+			}
+
+			if (value === null || value === undefined) return value;
+
+			if (Array.isArray(value)) {
+				return value.map((item: any) => obfuscateNode(key, item));
+			}
+
+			if (typeof value === 'symbol') return '********'; // technically impossible, but just in case
+
+			if (typeof value === 'object') {
+				for (const newKey in value) {
+					value[newKey] = obfuscateNode(newKey, value[newKey]);
+				}
+			}
+
+			return value;
+		};
+
+		const variablesCopy = JSON.parse(JSON.stringify(variables));
```

Comment:
It's cleaner to clone the object instance with { ...variables }

Reply:
Yeah, but that does a shallow copy. There are nested values then the references will still be attached. @juan871003 are there ever nested values in the variables object?